### PR TITLE
fix fp16 for tile op

### PR DESCRIPTION
**Pull request by enkilee**
### PR types

Others
### PR changes

APIs
### Describe

fix fp16 for tile op
- https://github.com/PaddlePaddle/Paddle/issues/50832 case 24
- https://github.com/PaddlePaddle/docs/pull/5643 update docs

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_tile_op.py [hunk 1]
```diff
@@ -366,6 +366,37 @@ def test_dygraph(self):
...
+            self.tile_wrapper, [data], out, x_init=[data_arr], place=place
+        )
+
+    def test_fp16(self):
+        paddle.enable_static()
+        places = [fluid.CPUPlace()]
+        if core.is_compiled_with_cuda():
+            places.append(fluid.CUDAPlace(0))
+        for p in places:
+            self.func(p)
```
Comment (Liyulingyue) [inline python/paddle/fluid/tests/unittests/test_tile_op.py]:
```suggestion
        places = [fluid.CUDAPlace(0)]
        if core.is_compiled_with_cuda():
            for p in places:
                self.func(p)
```

Comment (enkilee) [reply python/paddle/fluid/tests/unittests/test_tile_op.py]: Thx a lot

File: python/paddle/fluid/tests/unittests/test_tile_op.py [hunk 2]
```diff
@@ -366,6 +366,35 @@ def test_dygraph(self):
...
+        )
+        gradient_checker.double_grad_check_for_dygraph(
+            self.tile_wrapper, [data], out, x_init=[data_arr], place=place
+        )
+
+    def test_fp16(self):
+        paddle.enable_static()
+        places = [fluid.CPUPlace(0)]
+        for p in places:
+            self.func(p)
```
Comment (Liyulingyue) [inline python/paddle/fluid/tests/unittests/test_tile_op.py]: 需要判断当前是否属于GPU环境，然后仅在使用GPU的时候运行代码，不应该调用CPU

**Review by zhangting2020 — APPROVED** [review]
LGTM

**Pull request merged**